packages/multichain-account-service/src/MultichainAccountService.ts

```diff
@@ -359,12 +377,54 @@ export class MultichainAccountService {
     );
   }
 
+  /**
+   * Set basic functionality state and trigger alignment if enabled.
+   * When basic functionality is disabled, snap-based providers are disabled.
+   * When enabled, all snap providers are enabled and wallet alignment is triggered.
+   * EVM providers are never disabled as they're required for basic wallet functionality.
+   *
+   * @param enabled - Whether basic functionality is enabled.
+   */
+  async setBasicFunctionality(enabled: boolean): Promise<void> {
```

Comment:
IDK if we should have this as part of the public API 🤔

My initial idea was to have the "account provider wrapper" to "react" to some events (or maybe a runtime callback) and adapt each of their methods to return all accounts or no accounts (based on the "basic functionality flag").

Though, I haven't checked of this was done on both clients, so maybe it's not that easy to do